include/rapidcheck/state/Commands.hpp

```diff
@@ -1,9 +1,109 @@
 #pragma once
 
 #include "rapidcheck/detail/Results.h"
+#include "rapidcheck/state/gen/ParallelCommands.hpp"
+#include "rapidcheck/Assertions.h"
+#include <thread>
+#include <mutex>
+#include "rapidcheck/detail/Barrier.h"
 
 namespace rc {
 namespace state {
+namespace detail {
+
+
+template <typename Model, typename Cmd>
+struct CommandResult
+{
+  CommandResult(const std::shared_ptr<const Cmd>& command,
+                std::function<void(const Model&)>&& verifyFunc)
+    : command(command)
+    , verifyFunc(std::move(verifyFunc))
+  {
+  }
+
+  std::shared_ptr<const Cmd> command;
```

Comment:
Do we need this to be stored here? It's always the corresponding command in the sequence, right?